core/src/com/etheller/warsmash/parsers/jass/Jass2.java

```diff
@@ -328,187 +328,109 @@ public JUIEnvironment(final JassProgramVisitor jassProgramVisitor, final DataSou
-						final TriggerExecutionScope triggerScope) {
-					final UIFrame frame = arguments.get(0).visit(ObjectJassValueVisitor.<UIFrame>getInstance());
-					final FramePoint framePoint = arguments.get(1)
-							.visit(ObjectJassValueVisitor.<FramePoint>getInstance());
-					final double x = arguments.get(2).visit(RealJassValueVisitor.getInstance());
-					final double y = arguments.get(3).visit(RealJassValueVisitor.getInstance());
-
-					frame.setFramePointX(framePoint, GameUI.convertX(uiViewport, (float) x));
-					frame.setFramePointY(framePoint, GameUI.convertY(uiViewport, (float) y));
-					return null;
+				catch (final IOException e) {
+					throw new RuntimeException(e);
 				}
+				return BooleanJassValue.TRUE;
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("CreateSimpleFrame", (arguments, globalScope, triggerScope) -> {
+				final String templateName = arguments.get(0).visit(StringJassValueVisitor.getInstance());
+				final UIFrame ownerFrame = arguments.get(1).visit(ObjectJassValueVisitor.<UIFrame>getInstance());
+				final int createContext = arguments.get(2).visit(IntegerJassValueVisitor.getInstance());
+				final UIFrame simpleFrame = JUIEnvironment.this.gameUI.createSimpleFrame(templateName, ownerFrame, createContext);
+				return new HandleJassValue(frameHandleType, simpleFrame);
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("CreateFrame", (arguments, globalScope, triggerScope) -> {
+				final String templateName = arguments.get(0).visit(StringJassValueVisitor.getInstance());
+				final UIFrame ownerFrame = arguments.get(1).visit(ObjectJassValueVisitor.<UIFrame>getInstance());
+				final int priority = arguments.get(2).visit(IntegerJassValueVisitor.getInstance());
+				final int createContext = arguments.get(3).visit(IntegerJassValueVisitor.getInstance());
+				final UIFrame simpleFrame = JUIEnvironment.this.gameUI.createFrame(templateName, ownerFrame, priority, createContext);
+				return new HandleJassValue(frameHandleType, simpleFrame);
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("GetFrameByName", (arguments, globalScope, triggerScope) -> {
+				final String templateName = arguments.get(0).visit(StringJassValueVisitor.getInstance());
+				final int createContext = arguments.get(1).visit(IntegerJassValueVisitor.getInstance());
+				final UIFrame simpleFrame = JUIEnvironment.this.gameUI.getFrameByName(templateName, createContext);
+				return new HandleJassValue(frameHandleType, simpleFrame);
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("FrameSetAnchor", (arguments, globalScope, triggerScope) -> {
+				final String templateName = arguments.get(0).visit(StringJassValueVisitor.getInstance());
+				final int createContext = arguments.get(1).visit(IntegerJassValueVisitor.getInstance());
+				final UIFrame simpleFrame = JUIEnvironment.this.gameUI.getFrameByName(templateName, createContext);
```

Comment:
Git is failing to properly detect the changes between your work and the original, but I tried to analyze them in a more customized automated way and it complained that you have replaced the behavior of FrameSetAnchor here with the behavior of GetFrameByName from right above it, as if by mistake. It's probably a matter of some bunk copy paste human error, and we would not have noticed the issue in testing because these functions are currently not enabled from within custom map jass.